plugins/synaptics-rmi/fu-plugin-synaptics-rmi.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-plugin-vfuncs.h"
+
+#include "fu-synaptics-rmi-device.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_REQUIRES_QUIRK, FU_QUIRKS_PLUGIN);
```

Comment:
Do you still feel strongly that this needs to a `requires-quirk` instead of probing all the stuff that matches synaptics' vendor list?

Reply:
Could we do requires quirk and match on the vendor rather than vid/pid?

Reply:
Well the vendor would be synaptics wouldn't it?

Reply:
My point being if you just codify matching synaptics VID then no quirk file needed at all, and should be safe enough to probe.

Reply:
I've done:

    [DeviceInstanceId=HIDRAW\VEN_06CB]
    Plugin = synaptics_rmi

On the logic that this plugin won't get probed for every hidraw device on the system. It's much more efficient to filter in the engine, rather than create a FuDevice, create a derived object and copy all the properties of the udev device, find the vendor ID, tear down both objects.